supporter-product-data/src/main/scala/com/gu/lambdas/UpdateDynamoLambda.scala

```diff
@@ -0,0 +1,113 @@
+package com.gu.lambdas
+
+import com.amazonaws.services.lambda.runtime.Context
+import com.gu.lambdas.UpdateDynamoLambda.{batchSize, writeToDynamo}
+import com.gu.model.Stage
+import com.gu.model.dynamo.SupporterRatePlanItem
+import com.gu.model.states.UpdateDynamoState
+import com.gu.monitoring.SafeLogger
+import com.gu.monitoring.SafeLogger.Sanitizer
+import com.gu.services.{DynamoDBService, S3Service}
+import kantan.csv._
+import kantan.csv.ops._
+
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.concurrent.duration.DurationInt
+import scala.concurrent.{Await, Future}
+
+trait TimeOutCheck {
+  def timeRemainingMillis: Int
+}
+
+class ContextTimeOutCheck(context: Context) extends TimeOutCheck {
+  override def timeRemainingMillis = context.getRemainingTimeInMillis
+}
+
+class UpdateDynamoLambda extends Handler[UpdateDynamoState, UpdateDynamoState] {
+  override protected def handlerFuture(input: UpdateDynamoState, context: Context) = {
+    SafeLogger.info(s"Starting write to dynamo task for ${input.recordCount} records from ${input.filename}")
+    writeToDynamo(Stage.fromEnvironment, input, new ContextTimeOutCheck(context))
+  }
+}
+
+object UpdateDynamoLambda {
+  val batchSize = 10
+  val timeoutBufferInMillis = batchSize * 5 * 1000
+
+  def writeToDynamo(stage: Stage, state: UpdateDynamoState, timeOutCheck: TimeOutCheck): Future[UpdateDynamoState] = {
+    val csvStream = S3Service.streamFromS3(stage, state.filename)
+    val csvReader = csvStream.asCsvReader[SupporterRatePlanItem](rfc.withHeader)
+    val dynamoDBService = DynamoDBService(stage)
+
+    val unProcessed = getUnprocessedItems(csvReader, state.processedCount)
+
+    val validUnprocessed = unProcessed.filter { case (itemOrError, _) => itemOrError.isRight }
+    val invalidUnprocessed = unProcessed.filter { case (itemOrError, _) => itemOrError.isLeft }
+
+    if (invalidUnprocessed.nonEmpty)
```

Comment:
only do this on the first pass?
```suggestion
    if (invalidUnprocessed.nonEmpty && state.processedCount == 0)
```